packages/odf/components/attach-storage-storagesystem/lso-storageclass-dropdown.tsx

```diff
@@ -0,0 +1,145 @@
+import { NO_PROVISIONER } from '@odf/core/constants';
+import { useNodesData } from '@odf/core/hooks';
+import { StorageClassDropdown } from '@odf/core/modals/add-capacity/add-capacity-modal';
+import { PVsAvailableCapacity } from '@odf/core/modals/add-capacity/pvs-available-capacity';
+import { pvResource } from '@odf/core/resources';
+import { NodeData } from '@odf/core/types';
+import { checkArbiterCluster, checkFlexibleScaling } from '@odf/core/utils';
+import { getCephNodes } from '@odf/ocs/utils';
+import {
+  DeviceSet,
+  StorageClassResourceKind,
+  StorageClusterKind,
+  getName,
+} from '@odf/shared';
+import {
+  K8sResourceCommon,
+  useK8sWatchResource,
+} from '@openshift-console/dynamic-plugin-sdk';
+import { useTranslation } from 'react-i18next';
+import { Checkbox, FormGroup } from '@patternfly/react-core';
+import { createWizardNodeState, getDeviceSetReplica } from '../utils';
+import {
+  AttachStorageAction,
+  AttachStorageActionType,
+  AttachStorageFormState,
+} from './state';
+
+export const LSOStorageClassDropdown = ({
+  state,
+  dispatch,
+  storageCluster,
+  namespace,
+  existingStorageClasses,
+}: {
+  state: AttachStorageFormState;
+  dispatch: React.Dispatch<AttachStorageAction>;
+  storageCluster: StorageClusterKind;
+  namespace: string;
+  existingStorageClasses: StorageClassResourceKind[];
+}) => {
+  const { t } = useTranslation();
+  const [storageClass, setStorageClass] = React.useState(null);
+  const [pvData, pvLoaded, pvLoadError] =
+    useK8sWatchResource<K8sResourceCommon[]>(pvResource);
+  const [nodesData, nodesLoaded, nodesLoadError] = useNodesData();
+  const nodesError: boolean =
+    nodesLoadError || !(nodesData as []).length || !nodesLoaded;
+  const hasFlexibleScaling = checkFlexibleScaling(storageCluster);
+  const isArbiterEnabled: boolean = checkArbiterCluster(storageCluster);
+  const replica = getDeviceSetReplica(
+    isArbiterEnabled,
+    hasFlexibleScaling,
+    createWizardNodeState(getCephNodes(nodesData, namespace) as NodeData[])
+  );
+
+  const getStorageClassesNotInDeviceSet = React.useCallback(
+    (
+      storageClasses: StorageClassResourceKind[],
+      deviceSets: DeviceSet[]
+    ): string[] => {
+      const allStorageClasses = storageClasses.map(getName);
+      const deviceSetStorageClasses = deviceSets
+        .map((ds) => ds.dataPVCTemplate.spec.storageClassName)
+        .filter((name): name is string => Boolean(name));
+
+      return allStorageClasses.filter(
+        (scName) => !deviceSetStorageClasses.includes(scName)
+      );
+    },
+    [existingStorageClasses, storageCluster]
+  );
+
+  const preSelectionFilter = React.useCallback(
+    (storageClasses: StorageClassResourceKind[]) => {
+      const storageClassesNotInDeviceSet = getStorageClassesNotInDeviceSet(
+        storageClasses,
+        storageCluster?.spec?.storageDeviceSets || []
+      );
+      return storageClassesNotInDeviceSet[0];
+    },
+    [[getStorageClassesNotInDeviceSet]]
+  );
```

Comment:
`storageClassDropdownFilter` is already filtering all the existing StorageClasses, you directly select 0th of that list.
```suggestion
  const preSelectionFilter = React.useCallback(
    (storageClasses: StorageClassResourceKind[]) => storageClasses?.[0];
    },
    []
  );
```

Reply:
no need to filter again...